src/uucore/src/lib/mods/locale.rs

```diff
@@ -180,6 +189,52 @@ fn create_bundle(
     Ok(bundle)
 }
 
+/// Create a bundle from embedded locale files
+fn create_bundle_from_embedded(
+    locale: &LanguageIdentifier,
+    util_name: &str,
+) -> Result<FluentBundle<FluentResource>, LocalizationError> {
+    // Only support English from embedded files
+    if *locale != "en-US" {
+        return Err(LocalizationError::LocalesDirNotFound(
+            "Embedded locales only support en-US".to_string(),
+        ));
+    }
+
+    let embedded_locales = get_embedded_locales();
+    let locale_key = format!("{util_name}/en-US.ftl");
+
+    let ftl_content = embedded_locales.get(locale_key.as_str()).ok_or_else(|| {
+        LocalizationError::LocalesDirNotFound(format!("No embedded locale found for {util_name}"))
+    })?;
+
+    let resource = FluentResource::try_new(ftl_content.to_string()).map_err(
+        |(_partial_resource, mut errs): (FluentResource, Vec<ParserError>)| {
+            let first_err = errs.remove(0);
```

Comment:
Isn't it expensive to remove an element from a Vector? (vs just cloning errs[0], but I'm not even completely sure if you need to clone it anyway?)